datasource.go

```diff
@@ -35,10 +35,13 @@ type DataSource struct {
 	// Helper to read/write http headers
 	HTTPHeaders map[string]string `json:"-"`
 
-	JSONData       JSONData       `json:"jsonData,omitempty"`
-	SecureJSONData SecureJSONData `json:"secureJsonData,omitempty"`
+	JSONData         JSONData         `json:"jsonData,omitempty"`
+	SecureJSONData   SecureJSONData   `json:"secureJsonData,omitempty"`
+	SecureJSONFields SecureJSONFields `json:"secureJsonFields,omitempty"`
```

Comment:
`secureJsonFields` is a response field. You do not have to set it on the request and we do not need it in the response (the terraform provider only sets secure fields, it doesn't read them)